pkg/apis/application/v1alpha1/types.go

```diff
@@ -80,17 +87,21 @@ type ApplicationSource struct {
 	Directory *ApplicationSourceDirectory `json:"directory,omitempty" protobuf:"bytes,10,opt,name=directory"`
 	// ConfigManagementPlugin holds config management plugin specific options
 	Plugin *ApplicationSourcePlugin `json:"plugin,omitempty" protobuf:"bytes,11,opt,name=plugin"`
+	// Env is a list of environment variables passed to the tool
+	Env []Env `json:"env,omitempty" protobuf:"bytes,12,opt,name=env"`
```

Comment:
I think `Env` should be part of `ApplicationSourcePlugin`, not `ApplicationSource`

Reply:
this is very much open for discussion

Reply:
The `ApplicationSourcePlugin`, `ApplicationSourceKustomize`,  `ApplicationSourceKsonnet` etc were created to hold type-specific fields. Env is specific only for `ApplicationSourcePlugin`, right ?

Reply:
not sure, I'm just about to tabulate how each tool does "variables"

Reply:
| Tool | Global Variables |
|-|-|
| Directory | None |
| Helm | `values.yaml` | 
| Jsonnet | TLAs | 
| Ksonnet | TLAs |
| Kustomize | None? |
| Plugin | Env Var? |

Reply:
Looks like it is just Plugin. We've tried to have generic "parameters" for all tools but it quickly became too overloaded